<!DOCTYPE html>
<html>
<head>
  <title>CAGE data analysis</title>
  <meta charset="utf-8">
  <meta name="description" content="CAGE data analysis">
  <meta name="author" content="Dave Tang">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="./assets/css/ribbons.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>CAGE data analysis</h1>
    <h2>MODHEP workshop</h2>
    <p>Dave Tang<br/>PhD candidate</p>
  </hgroup>
    <a href="https://github.com/davetang/cage_r/zipball/gh-pages" class="example">
     Download
    </a>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="" id="slide-1" style="background:;">
  <article data-timings="">
    <!-- to publish run
library(slidify)
setwd("/Users/davetang/slidify_test/")
publish('cage_r', 'davetang')
-->

<style>
.title-slide {
  background-color: #FFFFFF;
}

pre {
 white-space: pre-wrap;       /* css-3 */
 white-space: -moz-pre-wrap;  /* Mozilla, since 1999 */
 white-space: -pre-wrap;      /* Opera 4-6 */
 white-space: -o-pre-wrap;    /* Opera 7 */
 word-wrap: break-word;       /* Internet Explorer 5.5+ */
}

</style>

<h2>Welcome!</h2>

<ul class = "build incremental">
<li>My name is Dave Tang; I was born in Hong Kong but raised in Papua New Guinea.</li>
<li>I&#39;m a PhD candidate in bioinformatics at the VU University Amsterdam.</li>
<li>I also maintain a <a href="http://davetang.org/muse">blog</a>, where I write about bioinformatics.</li>
<li>I&#39;m here to talk about analysing CAGE data.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <hgroup>
    <h2>About this presentation</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>I had a lot of difficulty deciding what to present in this talk.</li>
<li>I started off extremely ambiguous and believed that it is possible to show you how to analyse CAGE data in under two hours.</li>
<li>Then I realised that this isn&#39;t possible.</li>
<li>This presentation will provide an overview of analysing CAGE data</li>
<li>There are links on each slide for those who want to learn more.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h2>Slidify</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>These slides were made using an R package called <a href="http://slidify.org/index.html">Slidify</a>.</li>
<li>All the slides and the output are generated by R; it&#39;s like embedding analysis code in PowerPoint but also having PowerPoint perform the analysis.</li>
<li>This is the first time I&#39;ve used Slidify for a presentation.</li>
<li>These slides can be viewed at <a href="http://davetang.github.io/cage_r">http://davetang.github.io/cage_r</a></li>
<li>If you&#39;re interested in Slidify, have a look at <a href="http://davetang.org/muse/2014/12/12/making-slides-using-r/">http://davetang.org/muse/2014/12/12/making-slides-using-r/</a> and the links within.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <hgroup>
    <h2>Promoter</h2>
  </hgroup>
  <article data-timings="">
    <p>A promoter is a region of DNA that initiates transcription</p>

<p><img src="figure/25_1_2.jpg" alt="Eukaryotic promoter" style="width: 1000px;"/></p>

<p>Source: <a href="http://www.nature.com/scitable/topicpage/dna-transcription-426">http://www.nature.com/scitable/topicpage/dna-transcription-426</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h2>Cap Analysis Gene Expression (CAGE)</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="figure/cage_1.png" alt="CAGE protocol 1" style="width: 700px;"/></p>

<p>Image source: me.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h2>CAGE</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="figure/cage_2.png" alt="CAGE protocol 2" style="width: 700px;"/></p>

<p>For the full protocol see <a href="http://www.ncbi.nlm.nih.gov/pubmed/22362160">http://www.ncbi.nlm.nih.gov/pubmed/22362160</a>.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h2>Why do we perform CAGE?</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>Quantify the expression of transcripts</li>
<li>Study promoter structure and usage</li>
<li>Discover new promoters</li>
<li>Discover regulatory elements</li>
<li>For CAGE versus RNA-Seq see <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3975069/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3975069/</a></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <hgroup>
    <h2>A typical CAGE analysis pipeline</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>The output of CAGE is a set of short nucleotide sequences and their counts.</li>
<li>Typical steps in a CAGE analysis pipeline include:</li>
<li>Tag extraction</li>
<li>Quality control steps</li>
<li>Mapping tags to a reference genome</li>
<li>Tag clustering</li>
<li>Tag cluster annotation</li>
<li>Data analysis (e.g. differential expression analysis, co-expression networks, etc.)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-9" style="background:;">
  <hgroup>
    <h2>ENCODE CAGE data</h2>
  </hgroup>
  <article data-timings="">
    <p>Download these BAM files into some directory:</p>

<pre><code>path=http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeRikenCage
curl -O $path/wgEncodeRikenCageHelas3CellPapRawDataRep1.fastq.gz
curl -O $path/wgEncodeRikenCageHelas3CellPapAlnRep1.bam
curl -O $path/wgEncodeRikenCageHelas3CellPapAlnRep2.bam
curl -O $path/wgEncodeRikenCageHelas3CytosolPapAlnRep1.bam
curl -O $path/wgEncodeRikenCageHelas3CytosolPapAlnRep2.bam
curl -O $path/wgEncodeRikenCageHelas3NucleusPapAlnRep1.bam
curl -O $path/wgEncodeRikenCageHelas3NucleusPapAlnRep2.bam
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-10" style="background:;">
  <hgroup>
    <h2>Raw sequencing data</h2>
  </hgroup>
  <article data-timings="">
    <p>A <a href="http://en.wikipedia.org/wiki/FASTQ_format">fastq</a> file contains all the reads from the sequencer.</p>

<pre><code>gunzip -c wgEncodeRikenCageHelas3CellPapRawDataRep1.fastq.gz | head -4
@HWUSI-EAS566_0007:5:1:965:4705#0/1
NAGCAGCAGGGGAGAGTGTCATGGAGGCCTACGAGC
+HWUSI-EAS566_0007:5:1:965:4705#0/1
BYZZY\\[\[^^IXMVUUUX\]\\X[ZZ\]``^__^
</code></pre>

<p>For each read, there are four lines of information.</p>

<ol>
<li>The first line is the read name and contains information on the read</li>
<li>The second line is the raw sequence of the CAGE tag</li>
<li>The third line can be used to store optional notes</li>
<li>The fourth line denotes the quality of each base call, i.e. how sure the machine is that an A is an A.</li>
</ol>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-11" style="background:;">
  <hgroup>
    <h2>Tag extraction</h2>
  </hgroup>
  <article data-timings="">
    <pre><code>gunzip -c wgEncodeRikenCageHelas3CellPapRawDataRep1.fastq.gz | head -4
@HWUSI-EAS566_0007:5:1:965:4705#0/1
NAGCAGCAGGGGAGAGTGTCATGGAGGCCTACGAGC
+HWUSI-EAS566_0007:5:1:965:4705#0/1
BYZZY\\[\[^^IXMVUUUX\]\\X[ZZ\]``^__^

gunzip -c wgEncodeRikenCageHelas3CellPapRawDataRep1.fastq.gz | perl -nle &#39;if (($. - 2) % 4 == 0){ print substr($_,0,4)}&#39; | sort | uniq -c | sort -k1,1rn | head -4
25498964 TAGC
288819 TATC
154406 TCGT
117149 CAGC
</code></pre>

<p>The barcode for this library was TAGC. The <a href="http://hannonlab.cshl.edu/fastx_toolkit/">http://hannonlab.cshl.edu/fastx_toolkit/</a> contains tools for tag extraction. See also <a href="http://tagdust.sourceforge.net/">http://tagdust.sourceforge.net/</a>, which uses Hidden Markov Models for tag extraction (and performs quality control steps).</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-12" style="background:;">
  <hgroup>
    <h2>Quality control</h2>
  </hgroup>
  <article data-timings="">
    <pre><code>gunzip -c wgEncodeRikenCageHelas3CellPapRawDataRep1.fastq.gz | head -4
@HWUSI-EAS566_0007:5:1:965:4705#0/1
NAGCAGCAGGGGAGAGTGTCATGGAGGCCTACGAGC
+HWUSI-EAS566_0007:5:1:965:4705#0/1
BYZZY\\[\[^^IXMVUUUX\]\\X[ZZ\]``^__^
</code></pre>

<p>We would remove this read because it contains an ambiguous base call, especially since it&#39;s in the barcode. The quality scores range from 3 to 40:</p>

<pre>
3------------------------------------40

BCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefgh
</pre>

<p>Various different criteria to remove reads, e.g. remove reads with 10 base calls that have a quality less than 10, etc. The tool <a href="http://samstat.sourceforge.net/">http://samstat.sourceforge.net/</a> can be used to calculate and visualise read qualities.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-13" style="background:;">
  <hgroup>
    <h2>A note about quality scores</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">p &lt;- function(q){ return(10^{-q/10}) }
p(10)
</code></pre>

<pre><code>## [1] 0.1
</code></pre>

<pre><code class="r">plot(3:37, 1-p(3:37), xaxt=&#39;n&#39;, type=&#39;l&#39;, xlab=&#39;Quality score&#39;, ylab=&#39;Probability correct&#39;)
axis(1, 0:40); abline(h=0.9, v=10)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-1-1.png" title="plot of chunk unnamed-chunk-1" alt="plot of chunk unnamed-chunk-1" width="300px" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <hgroup>
    <h2>Mapping</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>High-throughput sequencing results in millions to billions of reads</li>
<li>A computational strategy known as &quot;indexing&quot; speeds up mapping algorithms.</li>
<li>It works like a book index; an index of a large DNA sequence allows one to rapidly find shorter sequences embedded within it. See <a href="http://www.ncbi.nlm.nih.gov/pubmed/19430453">How to map billions of short reads onto genomes</a>.</li>
<li>The confidence of mapping is indicated by the mapping quality, which uses the same scale as the base calling qualities.</li>
<li>Pipelines may remove reads that have a low mapping quality.</li>
<li>I have a post on <a href="http://davetang.org/muse/2011/09/14/mapping-qualities/">mapping qualities</a>.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <hgroup>
    <h2>SAM/BAM</h2>
  </hgroup>
  <article data-timings="">
    <p>A SAM/BAM file contains alignment information of a single read to a reference genome.</p>

<pre><code>samtools view wgEncodeRikenCageHelas3CellPapAlnRep1.bam | head -3
HWUSI-EAS566_0009:3:16:8371:2418#0|TAG  16  chr1    16213   1   27M *   0   0   GCCATGCTCTGACAGTCTCAGTTGCAC fefffffffffffffdfefddffdddd NM:i:0  MD:Z:27 XP:Z:~~~~-----000077777,,,,9998:
HWUSI-EAS566_0007:5:11:4550:12891#0|TAG 16  chr1    16445   17  27M *   0   0   AGTTTGAAAACCACTATTTTATGAACC fefffffdffffffeffdffffeefff NM:i:0  MD:Z:27 XP:Z:{{~~~~~~~~~~LO?~
HWUSI-EAS566_0007:5:63:12130:4370#0|TAG 0   chr1    16446   17  27M *   0   0   GTTTGAAAACCACTATTTTATGAACCA fffffdfffffdffffffffffefffe NM:i:0  MD:Z:27 XP:Z:~~~~~~~~~~~~}}L~
</code></pre>

<p>If you are confused about the second column, have a look at <a href="http://davetang.org/muse/2014/03/06/understanding-bam-flags/">http://davetang.org/muse/2014/03/06/understanding-bam-flags/</a>.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-16" style="background:;">
  <hgroup>
    <h2>CAGE defined transcription starting sites</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="figure/ctss.png" alt="CTSS" style="width: 700px;"/></p>

<p>Image source: me.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <hgroup>
    <h2>Tag cluster annotation</h2>
  </hgroup>
  <article data-timings="">
    <p>Intersect/overlap tag clusters to transcript annotations</p>

<p><img src="figure/ucsc_screenshot.png" alt="UCSC Genome Browser screenshot" style="width: 1000px" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-18" style="background:;">
  <hgroup>
    <h2>Intermission</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Any questions so far?</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-19" style="background:;">
  <hgroup>
    <h2>A few words on R before we begin</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>I have been using R on and off for a couple of years and it takes a while to get used to.</li>
<li>Honest confession: I am not very good with R (I keep a lot of <a href="http://davetang.org/muse/r/">documentation</a> to make up for this).</li>
<li>I think it is worthwhile learning because a lot of genomics analysis packages are available via the Bioconductor project.</li>
<li>By default, data loaded into R is stored into memory; the bigger your data, the more memory will be used.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-20" style="background:;">
  <hgroup>
    <h2>Bioconductor</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>From <a href="http://en.wikipedia.org/wiki/Bioconductor">Wikipedia</a>: Bioconductor is a free, open source and open development software project for the analysis and comprehension of genomic data generated by wet lab experiments in molecular biology.</li>
<li>Provides state of the art software to analyse various genomic datasets.</li>
<li>Has very well written guides, called vignettes, aimed at biologists.</li>
<li>To learn more take a look at these <a href="http://bioconductor.org/help/course-materials/">courses</a>, which are provided by the Bioconductor team.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-21" style="background:;">
  <hgroup>
    <h2>Loading data into R</h2>
  </hgroup>
  <article data-timings="">
    <p>Download the <a href="https://dl.dropboxusercontent.com/u/15251811/pnas_expression.txt">data</a> for this <a href="http://www.ncbi.nlm.nih.gov/pubmed/19088194">paper</a> (right-click on data and save as).</p>

<pre><code class="r">getwd()
</code></pre>

<pre><code>## [1] &quot;/Users/davetang/slidify_test&quot;
</code></pre>

<pre><code class="r">setwd(&quot;/Users/davetang/slidify_test/data/&quot;)
data &lt;- read.table(&quot;pnas_expression.txt&quot;, header=TRUE, sep=&quot;\t&quot;, stringsAsFactors = FALSE)
head(data, 5)
</code></pre>

<pre><code>##        ensembl_ID lane1 lane2 lane3 lane4 lane5 lane6 lane8  len
## 1 ENSG00000215696     0     0     0     0     0     0     0  330
## 2 ENSG00000215700     0     0     0     0     0     0     0 2370
## 3 ENSG00000215699     0     0     0     0     0     0     0 1842
## 4 ENSG00000215784     0     0     0     0     0     0     0 2393
## 5 ENSG00000212914     0     0     0     0     0     0     0  384
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-22" style="background:;">
  <hgroup>
    <h2>Working with data frames</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">class(data)
</code></pre>

<pre><code>## [1] &quot;data.frame&quot;
</code></pre>

<pre><code class="r">dim(data)
</code></pre>

<pre><code>## [1] 37435     9
</code></pre>

<pre><code class="r">data[1,]
</code></pre>

<pre><code>##        ensembl_ID lane1 lane2 lane3 lane4 lane5 lane6 lane8 len
## 1 ENSG00000215696     0     0     0     0     0     0     0 330
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-23" style="background:;">
  <hgroup>
    <h2>Removing a column</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">data_subset &lt;- data[,-9]
dim(data_subset)
</code></pre>

<pre><code>## [1] 37435     8
</code></pre>

<pre><code class="r">head(data_subset)
</code></pre>

<pre><code>##        ensembl_ID lane1 lane2 lane3 lane4 lane5 lane6 lane8
## 1 ENSG00000215696     0     0     0     0     0     0     0
## 2 ENSG00000215700     0     0     0     0     0     0     0
## 3 ENSG00000215699     0     0     0     0     0     0     0
## 4 ENSG00000215784     0     0     0     0     0     0     0
## 5 ENSG00000212914     0     0     0     0     0     0     0
## 6 ENSG00000212042     0     0     0     0     0     0     0
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-24" style="background:;">
  <hgroup>
    <h2>Naming the rows</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">row.names(data_subset) &lt;- data_subset[,1]
data_subset &lt;- data_subset[,-1]
head(data_subset)
</code></pre>

<pre><code>##                 lane1 lane2 lane3 lane4 lane5 lane6 lane8
## ENSG00000215696     0     0     0     0     0     0     0
## ENSG00000215700     0     0     0     0     0     0     0
## ENSG00000215699     0     0     0     0     0     0     0
## ENSG00000215784     0     0     0     0     0     0     0
## ENSG00000212914     0     0     0     0     0     0     0
## ENSG00000212042     0     0     0     0     0     0     0
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-25" style="background:;">
  <hgroup>
    <h2>Plotting the rowSums</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">plot(density(log2(rowSums(data_subset))), xlab=&quot;Gene expression (log2)&quot;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-6-1.png" alt="plot of chunk unnamed-chunk-6"> </p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-26" style="background:;">
  <hgroup>
    <h2>Independent filtering</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">dim(data_subset)
</code></pre>

<pre><code>## [1] 37435     7
</code></pre>

<pre><code class="r">data_subset &lt;- subset(data_subset, rowSums(data_subset)&gt;50)
dim(data_subset)
</code></pre>

<pre><code>## [1] 11539     7
</code></pre>

<p>See <a href="http://www.ncbi.nlm.nih.gov/pubmed/20460310">http://www.ncbi.nlm.nih.gov/pubmed/20460310</a>.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-27" style="background:;">
  <hgroup>
    <h2>Differential expression</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite(&quot;edgeR&quot;)
</code></pre>

<pre><code class="r">library(edgeR)
group &lt;- c(rep(&quot;Control&quot;,4), rep(&quot;Test&quot;,3))
d &lt;- DGEList(counts = data_subset, group=group)
#data normalisation
d &lt;- calcNormFactors(d, method=&quot;TMM&quot;)
d &lt;- estimateDisp(d)
et &lt;- exactTest(d)
summary(de &lt;- decideTestsDGE(et, p=0.05, adjust=&quot;BH&quot;))
</code></pre>

<pre><code>##    [,1]
## -1 2119
## 0  7215
## 1  2205
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-28" style="background:;">
  <hgroup>
    <h2>Differentially expressed genes</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">detags &lt;- rownames(d)[as.logical(de)]
plotSmear(et, de.tags=detags)
abline(h = c(-2, 2), col = &quot;blue&quot;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-10-1.png" title="plot of chunk unnamed-chunk-10" alt="plot of chunk unnamed-chunk-10" width="450px" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-29" style="background:;">
  <hgroup>
    <h2>Differentially expression genes</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">topTags(et, n=5)
</code></pre>

<pre><code>## Comparison of groups:  Test-Control 
##                    logFC    logCPM        PValue           FDR
## ENSG00000151503 5.789617  9.719792  0.000000e+00  0.000000e+00
## ENSG00000096060 4.977976  9.954372  0.000000e+00  0.000000e+00
## ENSG00000162772 3.289656  9.749036 6.589017e-263 2.534356e-259
## ENSG00000166451 4.658056  8.855415 8.833771e-254 2.548322e-250
## ENSG00000115648 2.571817 11.481226 3.797278e-236 8.763359e-233
</code></pre>

<pre><code class="r">data_subset[&#39;ENSG00000151503&#39;,]
</code></pre>

<pre><code>##                 lane1 lane2 lane3 lane4 lane5 lane6 lane8
## ENSG00000151503    35    35    49    59  3307  3439  1224
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-30" style="background:;">
  <hgroup>
    <h2>Saving the results</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">my_genes &lt;- data_subset[row.names(topTags(et, n=100)),]
head(my_genes, 3)
</code></pre>

<pre><code>##                 lane1 lane2 lane3 lane4 lane5 lane6 lane8
## ENSG00000151503    35    35    49    59  3307  3439  1224
## ENSG00000096060    65    79   105   113  3975  3727  1451
## ENSG00000162772   172   204   250   304  2972  3269  1112
</code></pre>

<pre><code class="r">write.table(my_genes, file = &quot;my_genes.tsv&quot;, quote = FALSE, sep = &quot;\t&quot;)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-31" style="background:;">
  <hgroup>
    <h2>Intermission 2</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Any questions or comments?</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-32" style="background:;">
  <hgroup>
    <h2>Correlation</h2>
  </hgroup>
  <article data-timings="">
    <p>See <a href="http://davetang.org/muse/2012/01/31/creating-a-correlation-matrix-with-r/">http://davetang.org/muse/2012/01/31/creating-a-correlation-matrix-with-r/</a></p>

<p><img src="figure/coexpression.png" alt="Co-expression" style="width: 700px;"/></p>

<p>Image source: adpated from <a href="http://www.ncbi.nlm.nih.gov/pubmed/20628352">http://www.ncbi.nlm.nih.gov/pubmed/20628352</a>.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-33" style="background:;">
  <hgroup>
    <h2>Pairwise comparisons</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">#a vs. b, a vs. c, b vs. c
choose(3,2)
</code></pre>

<pre><code>## [1] 3
</code></pre>

<pre><code class="r">choose(1000, 2)
</code></pre>

<pre><code>## [1] 499500
</code></pre>

<pre><code class="r">choose(30000, 2)
</code></pre>

<pre><code>## [1] 449985000
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-34" style="background:;">
  <hgroup>
    <h2>Pairwise comparisons</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">plot(1000:30000, log2(choose(1000:30000, 2)), type=&#39;l&#39;, xlab=&quot;n&quot;, ylab=&quot;Pairwise comparisons (log2)&quot;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-14-1.png" alt="plot of chunk unnamed-chunk-14"> </p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-35" style="background:;">
  <hgroup>
    <h2>Hierarchical clustering</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">h &lt;- hclust(dist(t(data_subset)))
plot(h)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-15-1.png" title="plot of chunk unnamed-chunk-15" alt="plot of chunk unnamed-chunk-15" width="450px" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-36" style="background:;">
  <hgroup>
    <h2>Distance measure</h2>
  </hgroup>
  <article data-timings="">
    <p>See <a href="http://davetang.org/muse/2013/08/15/distance-matrix-computation/">http://davetang.org/muse/2013/08/15/distance-matrix-computation/</a>.</p>

<pre><code class="r">dist(t(data_subset))
</code></pre>

<pre><code>##           lane1     lane2     lane3     lane4     lane5     lane6
## lane2  4874.042                                                  
## lane3 11119.401  7469.846                                        
## lane4 12085.221  8346.255  3337.774                              
## lane5 27151.883 24875.433 22593.797 22219.560                    
## lane6 27339.010 25050.592 22690.310 22325.354  3573.774          
## lane8 12519.767 15546.218 21483.780 22298.779 28479.141 28722.565
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-37" style="background:;">
  <hgroup>
    <h2>Heatmap</h2>
  </hgroup>
  <article data-timings="">
    <p>See <a href="http://davetang.org/muse/2010/12/06/making-a-heatmap-with-r/">http://davetang.org/muse/2010/12/06/making-a-heatmap-with-r/</a>.</p>

<pre><code class="r">#install.packages(&quot;gplots&quot;)
library(gplots)
heatmap.2(as.matrix(my_genes), scale=&quot;row&quot;, trace=&quot;none&quot;)
</code></pre>

<p><img src="assets/fig/unnamed-chunk-17-1.png" title="plot of chunk unnamed-chunk-17" alt="plot of chunk unnamed-chunk-17" width="400px" /></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-38" style="background:;">
  <hgroup>
    <h2>CAGEr</h2>
  </hgroup>
  <article data-timings="">
    <p>The CAGEr package available on Bioconductor provides various methods for analysing CAGE data. To install the CAGEr package:</p>

<pre><code class="r">source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite(&quot;CAGEr&quot;)
</code></pre>

<p>To load the CAGEr package:</p>

<pre><code class="r">library(CAGEr)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-39" style="background:;">
  <hgroup>
    <h2>What CAGEr can do</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>Quality control steps</li>
<li>CAGE normalisation</li>
<li>CAGE tag clustering</li>
<li>Analysis of promoter shape</li>
<li>Expression clustering</li>
<li>Analysis of promoter shifting</li>
<li>Importantly, we can use the CAGEr package to prepare our data for further downstream analyses.</li>
<li>For more information: <a href="http://davetang.org/muse/2013/04/16/cage-analysis-using-the-r-bioconductor-package-cage/">http://davetang.org/muse/2013/04/16/cage-analysis-using-the-r-bioconductor-package-cage/</a>.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-40" style="background:;">
  <hgroup>
    <h2>Some more resources</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>I highly recommend this book called <a href="http://shop.oreilly.com/product/0636920030157.do">Bioinformatics Data Skills</a>. I wrote a <a href="http://davetang.org/muse/2014/04/03/bioinformatics-data-skills/">book review</a> on it (on my blog of course).</li>
<li>My collection of links to various <a href="http://davetang.org/muse/primer/">introductory material related to bioinformatics</a>.</li>
<li>My collection of links to <a href="http://davetang.org/muse/r/">resources for learning about R</a>.</li>
<li>My collection of links to <a href="http://davetang.org/muse/learn/">courses or course material</a> related to bioinformatics.</li>
<li>Feel free <a href="mailto:me@davetang.org?subject=Hello%20from%20the%20MODHEP%20workshop!">to send me an email</a> any time.</li>
<li>Thank you for your attention!</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title=''>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='About this presentation'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='Slidify'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='Promoter'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='Cap Analysis Gene Expression (CAGE)'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='CAGE'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='Why do we perform CAGE?'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='A typical CAGE analysis pipeline'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='ENCODE CAGE data'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='Raw sequencing data'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='Tag extraction'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='Quality control'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='A note about quality scores'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title='Mapping'>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title='SAM/BAM'>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='CAGE defined transcription starting sites'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='Tag cluster annotation'>
         17
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=18 title='Intermission'>
         18
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=19 title='A few words on R before we begin'>
         19
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=20 title='Bioconductor'>
         20
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=21 title='Loading data into R'>
         21
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=22 title='Working with data frames'>
         22
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=23 title='Removing a column'>
         23
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=24 title='Naming the rows'>
         24
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=25 title='Plotting the rowSums'>
         25
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=26 title='Independent filtering'>
         26
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=27 title='Differential expression'>
         27
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=28 title='Differentially expressed genes'>
         28
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=29 title='Differentially expression genes'>
         29
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=30 title='Saving the results'>
         30
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=31 title='Intermission 2'>
         31
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=32 title='Correlation'>
         32
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=33 title='Pairwise comparisons'>
         33
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=34 title='Pairwise comparisons'>
         34
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=35 title='Hierarchical clustering'>
         35
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=36 title='Distance measure'>
         36
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=37 title='Heatmap'>
         37
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=38 title='CAGEr'>
         38
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=39 title='What CAGEr can do'>
         39
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=40 title='Some more resources'>
         40
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  
  <!-- MathJax: Fall back to local if CDN offline but local image fonts are not supported (saves >100MB) -->
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        inlineMath: [['$','$'], ['\\(','\\)']],
        processEscapes: true
      }
    });
  </script>
  <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  <!-- <script src="https://c328740.ssl.cf1.rackcdn.com/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script> -->
  <script>window.MathJax || document.write('<script type="text/x-mathjax-config">MathJax.Hub.Config({"HTML-CSS":{imageFont:null}});<\/script><script src="libraries/widgets/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"><\/script>')
</script>
<!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>