---
title       : Analysing CAGE data using R
subtitle    : MODHEP workshop
author      : Dave Tang
job         : PhD candidate
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      # 
widgets     : [mathjax]            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
knit        : slidify::knit2slides
github:
   user: davetang
   repo: cage_r
---

<!-- to publish run
publish('cage_r', 'davetang')
-->

<style>
.title-slide {
  background-color: #FFFFFF;
}
</style>

## Welcome!

> 1. My name is Dave
> 2. I'm a PhD student here at RIKEN
> 3. I'm here to (hopefully) show you how to analyse CAGE data using R. (Some of the methods can be applied to other high-throughput sequencing datasets.)

---

## I will be talking about

> 1. Estimating statistical significance of feature overlaps
> 2. Exploratory data analysis using R
> 3. Data normalisation and differential expression analysis
> 4. Visualising high-throughput sequencing data

---

## Aims

> 1. I want to explain why specific commands are being used (to the best of my ability), i.e. I don't simply want to teach you a "recipe".
> 2. This is simply because every dataset is different and you can't follow the same recipe each time, and you can interpret your results better.

---

## Slidify

> 1. These slides were made using an R package called Slidify. The output shown in the slides is directly generated by R. This is to aid reproducible research.
> 2. These slides can be viewed and downloaded from <http://davetang.github.io/cage_r>
> 3. If you're interested in Slidify, have a look at <http://davetang.org/muse/2014/12/12/making-slides-using-r/>

---

## Promoter

<img src="figure/25_1_2.jpg" alt="Eukaryotic promoter" style="width: 1000px;"/>

Source: <http://www.nature.com/scitable/topicpage/dna-transcription-426>

---

## Cap Analysis Gene Expression (CAGE)

<img src="figure/cage_1.png" alt="CAGE protocol 1" style="width: 700px;"/>

---

## CAGE

<img src="figure/cage_2.png" alt="CAGE protocol 2" style="width: 700px;"/>

---

## Bioconductor

1. From [Wikipedia](http://en.wikipedia.org/wiki/Bioconductor): Bioconductor is a free, open source and open development software project for the analysis and comprehension of genomic data generated by wet lab experiments in molecular biology.
2. Provides state of the art software to analyse various genomic datasets.
3. To learn more take a look at these [courses](http://bioconductor.org/help/course-materials/), which are provided by the Bioconductor team.

---

## CAGE data on Bioconductor


```r
#download and install
source("http://bioconductor.org/biocLite.R")
biocLite("FANTOM3and4CAGE")
```


```r
#looking at come CAGE data
library(FANTOM3and4CAGE)
data(FANTOMtissueCAGEhuman)
lung_group <- FANTOMtissueCAGEhuman[["lung"]]
head(lung_group)
```

```
##    chr    pos strand RCB-0702_WI-38 RCB-0098_A549 RCB-0465_Lu-130 lung
## 1 chr1 558799      +              0             1               0    0
## 2 chr1 559777      +              0             1               0    0
## 3 chr1 703878      -              0             1               0    0
## 4 chr1 703904      -              0             0               1    0
## 5 chr1 752741      -              0             0               1    0
## 6 chr1 752754      -              0             0               1    0
```

---

## CAGEr

The CAGEr package available on Bioconductor provides various methods for analysing CAGE data.


```r
source("http://bioconductor.org/biocLite.R")
biocLite("CAGEr")
```

Loading the libraries


```r
library(CAGEr)
```

---

## Random heatmap


```r
set.seed(31)
image(matrix(rnorm(100),nrow=10))
```

![plot of chunk unnamed-chunk-5](assets/fig/unnamed-chunk-5-1.png) 

---

## Clustering

![](figure/consensusClusters_expression_profiles_50.png)

---

## Formula

$$ a^2 + b^2 = c^2 $$

