---
title       : Analysing CAGE data using R
subtitle    : MODHEP workshop
author      : Dave Tang
job         : PhD candidate
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      # 
widgets     : [mathjax]            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
knit        : slidify::knit2slides
github:
   user: davetang
   repo: cage_r
---

<!-- to publish run
library(slidify)
setwd("slidify_test")
publish('cage_r', 'davetang')
-->

<style>
.title-slide {
  background-color: #FFFFFF;
}
</style>

## Welcome!

> * My name is Dave Tang; I was born in Hong Kong but raised in Papua New Guinea.
> * I'm a PhD candidate in bioinformatics at the VU University Amsterdam.
> * I'm here to show you how to analyse CAGE data using R in Japan.

---

## I will be talking about

> * Cap Analysis Gene Expression (CAGE)
> * Using R and Bioconductor packages to analyse CAGE data
> * Estimating statistical significance of feature overlaps
> * Data normalisation and differential expression analysis
> * Visualising CAGE data

---

## About this presentation

> * I structured this presentation as if I'm presenting it to myself from five years ago (when I first started analysing CAGE data).
> * Therefore, I assume that you don't know anything about CAGE or the analysis of CAGE data.
> * Please interrupt me as soon as something is not clear! (Otherwise I'm not sure if I have enough material for two hours.)

---

## Slidify

> * These slides were made using an R package called [Slidify](http://slidify.org/index.html).
> * All the slides and the output are generated by R; the analysis is run when making the slides.
> * This is the first time I've used Slidify for a presentation.
> * These slides can be viewed and downloaded at <http://davetang.github.io/cage_r>
> * If you're interested in Slidify, have a look at <http://davetang.org/muse/2014/12/12/making-slides-using-r/> and the links within.

---

## Promoter

<img src="figure/25_1_2.jpg" alt="Eukaryotic promoter" style="width: 1000px;"/>

Source: <http://www.nature.com/scitable/topicpage/dna-transcription-426>

---

## Cap Analysis Gene Expression (CAGE)

<img src="figure/cage_1.png" alt="CAGE protocol 1" style="width: 700px;"/>

Image source: me.

---

## CAGE

<img src="figure/cage_2.png" alt="CAGE protocol 2" style="width: 700px;"/>

Image source: me.

---

## CAGE

> * The output of CAGE is a set of short nucleotide sequences and their counts.
> * Typical steps in a CAGE analysis pipeline include:
> * Tag extraction
> * Quality control of tags
> * Mapping CAGE tags to a reference genome
> * Tag clustering
> * Tag cluster annotation
> * Data analysis (e.g. differential expression analysis, co-expression networks, etc.)

---

## Why do we perform CAGE?

> * Discover rare transcripts
> * Quantify the expression of genes (especially comparing different conditions)
> * Study promoter structure and usage
> * Discover regulatory elements

---

## A few words on R

> * I have been using R on and off for a couple of years and it took a while to get used to.
> * Honest confession: I personally believe that I'm not very good with R (I keep a lot of documentation to make up for this).
> * You should learn it because a lot of the analysis packages for genomics data are provided via Bioconductor.

---

## Bioconductor

> * From [Wikipedia](http://en.wikipedia.org/wiki/Bioconductor): Bioconductor is a free, open source and open development software project for the analysis and comprehension of genomic data generated by wet lab experiments in molecular biology.
> * Provides state of the art software to analyse various genomic datasets.
> * Has well written guides for biologists!
> * To learn more take a look at these [courses](http://bioconductor.org/help/course-materials/), which are provided by the Bioconductor team.

---

## Some basics in R

```{r}
class(iris)
head(iris)
```

---

## Subsetting

```{r}
head(subset(iris, Sepal.Width > 3.5))
```

---

## Getting help in R

```{r eval=FALSE}
#returns all the functions with 'table' in the name
apropos('table')
#opens the documentation for the function 'table'
?table
#read manual
browseVignettes('CAGEr')
```

---

## CAGEr

The CAGEr package available on Bioconductor provides various methods for analysing CAGE data. To install the CAGEr package:

```{r eval=FALSE}
source("http://bioconductor.org/biocLite.R")
biocLite("CAGEr")
```

To load the CAGEr package:

```{r eval=FALSE}
library(CAGEr)
```

For more information: <http://davetang.org/muse/2013/04/16/cage-analysis-using-the-r-bioconductor-package-cage/>.

---

## What CAGEr can do

> * CAGE normalisation
> * CAGE tag clustering
> * Analysis of promoter shape
> * Expression clustering
> * Analysis of promoter shifting
> * Importantly, we can use the CAGEr package to prepare our data for further downstream analyses

---

## A SAM/BAM file

---

## CAGE data on Bioconductor

```{r eval=FALSE}
#download and install
source("http://bioconductor.org/biocLite.R")
biocLite("FANTOM3and4CAGE")
```

```{r eval=FALSE}
#looking at come CAGE data
library(FANTOM3and4CAGE)
data(FANTOMtissueCAGEhuman)
lung_group <- FANTOMtissueCAGEhuman[["lung"]]
head(lung_group)
```

---

## The need for normalisation

```{r}
control_1 <- rep(10, 50)
control_2 <- rep(10, 50)
patient_1 <- c(rep(20, 25),rep(0,25))
patient_2 <- c(rep(20, 25),rep(0,25))
df <- data.frame(c1=control_1, c2=control_2, p1=patient_1, p2=patient_2)
head(df, 2)
tail(df, 2)
```

---

## The need for normalisation

```{r echo=FALSE}
library(gplots)
heatmap.2(x = as.matrix(df),
          trace="none",
          dendrogram="none")
```

---

## Differential expression without normalisation

```{r, message=FALSE}
library(edgeR)
group <- c('control','control','patient','patient')
d <- DGEList(counts=df, group=group)
d <- estimateCommonDisp(d)
de <- exactTest(d)
table(p.adjust(de$table$PValue, method="BH")<0.05)
```

---

## Differential expression with normalisation

```{r, message=FALSE}
TMM <- calcNormFactors(d, method="TMM")
TMM$samples
TMM <- estimateCommonDisp(TMM)
TMM <- exactTest(TMM)
table(p.adjust(TMM$table$PValue, method="BH")<0.05)
```

---

## Genomic ranges

```{r}
library(IRanges)
ir <- IRanges(5,10)
ir
start(ir)
end(ir)
width(ir)
```

---

## Statistical significance of feature overlaps

Check out my blog post <http://davetang.org/muse/2014/11/07/using-genometricorr-package/>

```{r eval=FALSE}
install.packages('GenometriCorr',
                 repos='http://genometricorr.sourceforge.net/R/',
                 type='source')

cpg <- read.table(url("http://quinlanlab.cs.virginia.edu/cshl2013/cpg.bed"),
                  header=F,
                  sep="\t",
                  stringsAsFactors = F)

chr <- read.table(url("http://quinlanlab.cs.virginia.edu/cshl2013/hesc.chromHmm.bed"),
                  header=F,
                  sep="\t",
                  stringsAsFactors = F)
```

---

## Interpreting expression data

> * Which clustering techniques are useful for interpreting expression data?
> * A common computational approach is hierarchical clustering
> * SOMs have a number of features that make them particularly well suited to clustering and analysis of expression patterns.

---

## Clustering

![](figure/consensusClusters_expression_profiles_50.png)

---

## Visualisation

<http://davetang.org/muse/2013/10/03/using-gviz/>

---

## To be continued

