<!DOCTYPE html>
<html>
<head>
  <title>Analysing CAGE data using R</title>
  <meta charset="utf-8">
  <meta name="description" content="Analysing CAGE data using R">
  <meta name="author" content="Dave Tang">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="./assets/css/ribbons.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="libraries/frameworks/io2012/js/slides" 
    src="libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>Analysing CAGE data using R</h1>
    <h2>MODHEP workshop</h2>
    <p>Dave Tang<br/>PhD candidate</p>
  </hgroup>
    <a href="https://github.com/davetang/cage_r/zipball/gh-pages" class="example">
     Download
    </a>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="" id="slide-1" style="background:;">
  <article data-timings="">
    <!-- to publish run
library(slidify)
setwd("slidify_test")
publish('cage_r', 'davetang')
-->

<style>
.title-slide {
  background-color: #FFFFFF;
}

pre {
 white-space: pre-wrap;       /* css-3 */
 white-space: -moz-pre-wrap;  /* Mozilla, since 1999 */
 white-space: -pre-wrap;      /* Opera 4-6 */
 white-space: -o-pre-wrap;    /* Opera 7 */
 word-wrap: break-word;       /* Internet Explorer 5.5+ */
}

</style>

<h2>Welcome!</h2>

<ul class = "build incremental">
<li>My name is Dave Tang; I was born in Hong Kong but raised in Papua New Guinea.</li>
<li>I&#39;m a PhD candidate in bioinformatics at the VU University Amsterdam.</li>
<li>I also maintain a <a href="http://davetang.org/muse">blog</a>, where I write about bioinformatics.</li>
<li>I&#39;m here to show you how to analyse CAGE data using R in Japan.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <hgroup>
    <h2>About this presentation</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>I had a lot of difficulty deciding what to present in this talk.</li>
<li>I started off extremely ambiguous and believed that it is possible to show you how to analyse CAGE data in under two hours.</li>
<li>I realised that this isn&#39;t possible.</li>
<li>I finally decided that this presentation will be about the concepts of CAGE analysis and I will provide links on each slide for those who want to learn more.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h2>I will be talking about</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>Cap Analysis Gene Expression (CAGE)</li>
<li>Using R and Bioconductor packages to analyse CAGE data</li>
<li>Estimating statistical significance of feature overlaps</li>
<li>Data normalisation and differential expression analysis</li>
<li>Data visualistion of CAGE data in R</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <hgroup>
    <h2>Slidify</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>These slides were made using an R package called <a href="http://slidify.org/index.html">Slidify</a>.</li>
<li>All the slides and the output are generated by R; the analysis is run when making the slides.</li>
<li>This is the first time I&#39;ve used Slidify for a presentation.</li>
<li>These slides can be viewed and downloaded at <a href="http://davetang.github.io/cage_r">http://davetang.github.io/cage_r</a></li>
<li>If you&#39;re interested in Slidify, have a look at <a href="http://davetang.org/muse/2014/12/12/making-slides-using-r/">http://davetang.org/muse/2014/12/12/making-slides-using-r/</a> and the links within.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h2>Promoter</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="figure/25_1_2.jpg" alt="Eukaryotic promoter" style="width: 1000px;"/></p>

<p>Source: <a href="http://www.nature.com/scitable/topicpage/dna-transcription-426">http://www.nature.com/scitable/topicpage/dna-transcription-426</a></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h2>Cap Analysis Gene Expression (CAGE)</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="figure/cage_1.png" alt="CAGE protocol 1" style="width: 700px;"/></p>

<p>Image source: me.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h2>CAGE</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="figure/cage_2.png" alt="CAGE protocol 2" style="width: 700px;"/></p>

<p>Image source: me.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <hgroup>
    <h2>CAGE</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>The output of CAGE is a set of short nucleotide sequences and their counts.</li>
<li>Typical steps in a CAGE analysis pipeline include:</li>
<li>Tag extraction</li>
<li>Quality control steps</li>
<li>Mapping tags to a reference genome</li>
<li>Tag clustering</li>
<li>Tag cluster annotation</li>
<li>Data analysis (e.g. differential expression analysis, co-expression networks, etc.)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-9" style="background:;">
  <hgroup>
    <h2>Why do we perform CAGE?</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>Discover rare transcripts</li>
<li>Quantify the expression of transcripts</li>
<li>Study promoter structure and usage</li>
<li>Discover regulatory elements</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-10" style="background:;">
  <hgroup>
    <h2>ENCODE CAGE data</h2>
  </hgroup>
  <article data-timings="">
    <p>Download these BAM files into some directory:</p>

<pre><code>path=http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeRikenCage
curl -O $path/wgEncodeRikenCageHelas3CellPapRawDataRep1.fastq.gz
curl -O $path/wgEncodeRikenCageHelas3CellPapAlnRep1.bam
curl -O $path/wgEncodeRikenCageHelas3CellPapAlnRep2.bam
curl -O $path/wgEncodeRikenCageHelas3CytosolPapAlnRep1.bam
curl -O $path/wgEncodeRikenCageHelas3CytosolPapAlnRep2.bam
curl -O $path/wgEncodeRikenCageHelas3NucleusPapAlnRep1.bam
curl -O $path/wgEncodeRikenCageHelas3NucleusPapAlnRep2.bam
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-11" style="background:;">
  <hgroup>
    <h2>Raw CAGE data</h2>
  </hgroup>
  <article data-timings="">
    <p>A fastq file contains all the reads from the sequencer.</p>

<pre><code>gunzip -c wgEncodeRikenCageHelas3CellPapRawDataRep1.fastq.gz | head -5
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-12" style="background:;">
  <hgroup>
    <h2>SAM/BAM</h2>
  </hgroup>
  <article data-timings="">
    <p>A SAM/BAM file contains alignment information of a read to a reference genome.</p>

<pre><code>samtools view wgEncodeRikenCageHelas3CellPapAlnRep1.bam | head -5
HWUSI-EAS566_0009:3:16:8371:2418#0|TAG  16  chr1    16213   1   27M *   0   0   GCCATGCTCTGACAGTCTCAGTTGCAC fefffffffffffffdfefddffdddd NM:i:0  MD:Z:27 XP:Z:~~~~-----000077777,,,,9998:
HWUSI-EAS566_0007:5:11:4550:12891#0|TAG 16  chr1    16445   17  27M *   0   0   AGTTTGAAAACCACTATTTTATGAACC fefffffdffffffeffdffffeefff NM:i:0  MD:Z:27 XP:Z:{{~~~~~~~~~~LO?~
HWUSI-EAS566_0007:5:63:12130:4370#0|TAG 0   chr1    16446   17  27M *   0   0   GTTTGAAAACCACTATTTTATGAACCA fffffdfffffdffffffffffefffe NM:i:0  MD:Z:27 XP:Z:~~~~~~~~~~~~}}L~
HWUSI-EAS566_0007:5:107:18395:14990#0|TAG   16  chr1    16448   18  27M *   0   0   TTGAAAACCACTATTTTATGAACCAAA edcddbeeee`fdfffdaddaeeeeea NM:i:1  MD:Z:26G    XP:Z:{{~~~~~~~~~~B--J
HWUSI-EAS566_0007:5:39:15300:6753#0|TAG 16  chr1    16580   1   27M *   0   0   AAGGTGGCCTCAAAGAGCTGCTCCCAC dffffffffeafffffffaffffdfef NM:i:1  MD:Z:0C26   XP:Z:M9BC~~~~~~~~~~}}
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-13" style="background:;">
  <hgroup>
    <h2>A few words on R</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>I have been using R on and off for a couple of years and it took a while to get used to.</li>
<li>Honest confession: I personally believe that I&#39;m not very good with R (I keep a lot of documentation to make up for this).</li>
<li>You should learn it because a lot of the analysis packages for genomics data are provided via Bioconductor.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <hgroup>
    <h2>Bioconductor</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>From <a href="http://en.wikipedia.org/wiki/Bioconductor">Wikipedia</a>: Bioconductor is a free, open source and open development software project for the analysis and comprehension of genomic data generated by wet lab experiments in molecular biology.</li>
<li>Provides state of the art software to analyse various genomic datasets.</li>
<li>Has well written guides for biologists!</li>
<li>To learn more take a look at these <a href="http://bioconductor.org/help/course-materials/">courses</a>, which are provided by the Bioconductor team.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <hgroup>
    <h2>Some basics in R</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">class(iris)
</code></pre>

<pre><code>## [1] &quot;data.frame&quot;
</code></pre>

<pre><code class="r">head(iris)
</code></pre>

<pre><code>##   Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 1          5.1         3.5          1.4         0.2  setosa
## 2          4.9         3.0          1.4         0.2  setosa
## 3          4.7         3.2          1.3         0.2  setosa
## 4          4.6         3.1          1.5         0.2  setosa
## 5          5.0         3.6          1.4         0.2  setosa
## 6          5.4         3.9          1.7         0.4  setosa
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-16" style="background:;">
  <hgroup>
    <h2>Subsetting</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">head(subset(iris, Sepal.Width &gt; 3.5))
</code></pre>

<pre><code>##    Sepal.Length Sepal.Width Petal.Length Petal.Width Species
## 5           5.0         3.6          1.4         0.2  setosa
## 6           5.4         3.9          1.7         0.4  setosa
## 11          5.4         3.7          1.5         0.2  setosa
## 15          5.8         4.0          1.2         0.2  setosa
## 16          5.7         4.4          1.5         0.4  setosa
## 17          5.4         3.9          1.3         0.4  setosa
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <hgroup>
    <h2>Getting help in R</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">#returns all the functions with &#39;table&#39; in the name
apropos(&#39;table&#39;)
#opens the documentation for the function &#39;table&#39;
?table
#read manual
browseVignettes(&#39;CAGEr&#39;)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-18" style="background:;">
  <hgroup>
    <h2>CAGEr</h2>
  </hgroup>
  <article data-timings="">
    <p>The CAGEr package available on Bioconductor provides various methods for analysing CAGE data. To install the CAGEr package:</p>

<pre><code class="r">source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite(&quot;CAGEr&quot;)
</code></pre>

<p>To load the CAGEr package:</p>

<pre><code class="r">library(CAGEr)
</code></pre>

<p>For more information: <a href="http://davetang.org/muse/2013/04/16/cage-analysis-using-the-r-bioconductor-package-cage/">http://davetang.org/muse/2013/04/16/cage-analysis-using-the-r-bioconductor-package-cage/</a>.</p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-19" style="background:;">
  <hgroup>
    <h2>What CAGEr can do</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>CAGE normalisation</li>
<li>CAGE tag clustering</li>
<li>Analysis of promoter shape</li>
<li>Expression clustering</li>
<li>Analysis of promoter shifting</li>
<li>Importantly, we can use the CAGEr package to prepare our data for further downstream analyses</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-20" style="background:;">
  <hgroup>
    <h2>Loading BAM files</h2>
  </hgroup>
  <article data-timings="">
    <p>First we need to set the working directory to where the BAM files are:</p>

<pre><code class="r">setwd(&quot;/Users/davetang/tmp/cage&quot;)
bam_file &lt;- list.files(&quot;.&quot;, full.names=T, pattern=&quot;*.bam&quot;)
bam_file
cage_bam &lt;- new(&quot;CAGEset&quot;, genomeName = &quot;BSgenome.Hsapiens.UCSC.hg19&quot;, inputFiles = bam_file, inputFilesType = &#39;bam&#39;, sampleLabels = c(&#39;cell_1&#39;,&#39;cell_2&#39;,&#39;cytosol_1&#39;,&#39;cytosol_2&#39;,&#39;nucleus_1&#39;,&#39;nucleus_2&#39;))
cage_bam
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-21" style="background:;">
  <hgroup>
    <h2>Tag clustering</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">getCTSS(cage_bam, mappingQualityThreshold=10)
normalizeTagCount(cage_bam, method = &quot;simpleTpm&quot;)
clusterCTSS(object = cage_bam, threshold = 1, thresholdIsTpm = TRUE, nrPassThreshold = 1, method = &quot;distclu&quot;, maxDist = 20, removeSingletons = TRUE, keepSingletonsAbove = 5, useMulticore = T, nrCores = 6)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-22" style="background:;">
  <hgroup>
    <h2>CAGE data on Bioconductor</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">#download and install
source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite(&quot;FANTOM3and4CAGE&quot;)
</code></pre>

<pre><code class="r">#looking at come CAGE data
library(FANTOM3and4CAGE)
data(FANTOMtissueCAGEhuman)
lung_group &lt;- FANTOMtissueCAGEhuman[[&quot;lung&quot;]]
head(lung_group)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-23" style="background:;">
  <hgroup>
    <h2>The need for normalisation</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">control_1 &lt;- rep(10, 50)
control_2 &lt;- rep(10, 50)
patient_1 &lt;- c(rep(20, 25),rep(0,25))
patient_2 &lt;- c(rep(20, 25),rep(0,25))
df &lt;- data.frame(c1=control_1, c2=control_2, p1=patient_1, p2=patient_2)
head(df, 2)
</code></pre>

<pre><code>##   c1 c2 p1 p2
## 1 10 10 20 20
## 2 10 10 20 20
</code></pre>

<pre><code class="r">tail(df, 2)
</code></pre>

<pre><code>##    c1 c2 p1 p2
## 49 10 10  0  0
## 50 10 10  0  0
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-24" style="background:;">
  <hgroup>
    <h2>The need for normalisation</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="assets/fig/unnamed-chunk-11-1.png" alt="plot of chunk unnamed-chunk-11"> </p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-25" style="background:;">
  <hgroup>
    <h2>Differential expression without normalisation</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">library(edgeR)
group &lt;- c(&#39;control&#39;,&#39;control&#39;,&#39;patient&#39;,&#39;patient&#39;)
d &lt;- DGEList(counts=df, group=group)
d &lt;- estimateCommonDisp(d)
de &lt;- exactTest(d)
table(p.adjust(de$table$PValue, method=&quot;BH&quot;)&lt;0.05)
</code></pre>

<pre><code>## 
## TRUE 
##   50
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-26" style="background:;">
  <hgroup>
    <h2>Differential expression with normalisation</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">TMM &lt;- calcNormFactors(d, method=&quot;TMM&quot;)
TMM$samples
</code></pre>

<pre><code>##      group lib.size norm.factors
## c1 control      500    0.7071068
## c2 control      500    0.7071068
## p1 patient      500    1.4142136
## p2 patient      500    1.4142136
</code></pre>

<pre><code class="r">TMM &lt;- estimateCommonDisp(TMM)
TMM &lt;- exactTest(TMM)
table(p.adjust(TMM$table$PValue, method=&quot;BH&quot;)&lt;0.05)
</code></pre>

<pre><code>## 
## FALSE  TRUE 
##    25    25
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-27" style="background:;">
  <hgroup>
    <h2>Genomic ranges</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>The coordinate system for describing the location of features on a genome is through genomic ranges.</li>
<li>A genomic range indicates the chromosome and position a particular feature is on. For example, HBA1 is on chr16:226650-227521.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-28" style="background:;">
  <hgroup>
    <h2>Genomic ranges in Bioconductor</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">library(IRanges)
ir &lt;- IRanges(5,10)
ir
</code></pre>

<pre><code>## IRanges of length 1
##     start end width
## [1]     5  10     6
</code></pre>

<pre><code class="r">start(ir)
</code></pre>

<pre><code>## [1] 5
</code></pre>

<pre><code class="r">end(ir)
</code></pre>

<pre><code>## [1] 10
</code></pre>

<pre><code class="r">width(ir)
</code></pre>

<pre><code>## [1] 6
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-29" style="background:;">
  <hgroup>
    <h2>Statistical significance of feature overlaps</h2>
  </hgroup>
  <article data-timings="">
    <p>Check out my blog post <a href="http://davetang.org/muse/2014/11/07/using-genometricorr-package/">http://davetang.org/muse/2014/11/07/using-genometricorr-package/</a></p>

<pre><code class="r">install.packages(&#39;GenometriCorr&#39;,
                 repos=&#39;http://genometricorr.sourceforge.net/R/&#39;,
                 type=&#39;source&#39;)

cpg &lt;- read.table(url(&quot;http://quinlanlab.cs.virginia.edu/cshl2013/cpg.bed&quot;),
                  header=F,
                  sep=&quot;\t&quot;,
                  stringsAsFactors = F)

chr &lt;- read.table(url(&quot;http://quinlanlab.cs.virginia.edu/cshl2013/hesc.chromHmm.bed&quot;),
                  header=F,
                  sep=&quot;\t&quot;,
                  stringsAsFactors = F)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-30" style="background:;">
  <hgroup>
    <h2>Interpreting expression data</h2>
  </hgroup>
  <article data-timings="">
    <ul class = "build incremental">
<li>Which clustering techniques are useful for interpreting expression data?</li>
<li>A common computational approach is hierarchical clustering</li>
<li>SOMs have a number of features that make them particularly well suited to clustering and analysis of expression patterns.</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-31" style="background:;">
  <hgroup>
    <h2>Expression clustering</h2>
  </hgroup>
  <article data-timings="">
    <p><img src="figure/consensusClusters_expression_profiles_50.png" alt=""></p>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-32" style="background:;">
  <hgroup>
    <h2>Using BEDTools</h2>
  </hgroup>
  <article data-timings="">
    
  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title=''>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='About this presentation'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='I will be talking about'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='Slidify'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='Promoter'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='Cap Analysis Gene Expression (CAGE)'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='CAGE'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='CAGE'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='Why do we perform CAGE?'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='ENCODE CAGE data'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='Raw CAGE data'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='SAM/BAM'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='A few words on R'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title='Bioconductor'>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title='Some basics in R'>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='Subsetting'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='Getting help in R'>
         17
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=18 title='CAGEr'>
         18
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=19 title='What CAGEr can do'>
         19
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=20 title='Loading BAM files'>
         20
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=21 title='Tag clustering'>
         21
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=22 title='CAGE data on Bioconductor'>
         22
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=23 title='The need for normalisation'>
         23
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=24 title='The need for normalisation'>
         24
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=25 title='Differential expression without normalisation'>
         25
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=26 title='Differential expression with normalisation'>
         26
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=27 title='Genomic ranges'>
         27
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=28 title='Genomic ranges in Bioconductor'>
         28
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=29 title='Statistical significance of feature overlaps'>
         29
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=30 title='Interpreting expression data'>
         30
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=31 title='Expression clustering'>
         31
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=32 title='Using BEDTools'>
         32
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  
  <!-- MathJax: Fall back to local if CDN offline but local image fonts are not supported (saves >100MB) -->
  <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        inlineMath: [['$','$'], ['\\(','\\)']],
        processEscapes: true
      }
    });
  </script>
  <script type="text/javascript" src="http://cdn.mathjax.org/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
  <!-- <script src="https://c328740.ssl.cf1.rackcdn.com/mathjax/2.0-latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script> -->
  <script>window.MathJax || document.write('<script type="text/x-mathjax-config">MathJax.Hub.Config({"HTML-CSS":{imageFont:null}});<\/script><script src="libraries/widgets/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML"><\/script>')
</script>
<!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>